### PR TITLE
cmds: update to new interface of phfs

### PR DESCRIPTION
**Pull request by HBuczynski**
This PR should be started reviewing when [this one](https://github.com/phoenix-rtos/plo/pull/55) will be merged and `cmds` branch will be rebased.

### CODE REVIEW

File: cmd.c [hunk 1]
```diff
@@ -384,89 +375,81 @@ void cmd_write(char *s)
...
 	}
 
-	/* Parse source parameters */
-	if (cmd_checkDev(cmdArgs[argsID], &phfses[0].dn) < 0)
-		return;
-
-	if (cmd_parseDev(&phfses[0], cmdArgs, &argsID, cmdArgsc) < 0) {
-		phfs_close(phfses[0].dn, phfses[0].handle);
+	if (cmd_parseDev(&h[0], &offs[0], &sz[0], args, &argsID, argsc) < 0) {
+		phfs_close(h[0]);
```
Comment (agkaminski) [inline cmd.c:435]: If we failed to parseDev do we need to close the handler? I would assume that it shouldn't have been opened in that case, so there is nothing to close.

File: cmd.c [hunk 2]
```diff
@@ -843,6 +843,75 @@ void cmd_syspage(char *s)
...
+	for (i = 0; argsc < 5; ++i) {
+		if (cmd_getnext(s, &pos, ". \t", NULL, args[i], sizeof(args[i])) == NULL || *args[i] == 0)
+			break;
+		argsc++;
+	}
+
+	if (argsc < 3 || argsc > 4) {
+		plostd_printf(ATTR_ERROR, "\nWrong arguments!!\n");
+		return;
+	}
```
Comment (agkaminski) [inline cmd.c:861]: First we parse arguments, then we check how many there were? I think that `if` and `for` should switch places.

File: cmd.c [hunk 3]
```diff
@@ -843,6 +843,75 @@ void cmd_syspage(char *s)
...
+	unsigned int pos = 0;
+
+	cmd_skipblanks(s, &pos, DEFAULT_BLANKS);
+	s += pos;
+
+	if (*s) {
+		plostd_printf(ATTR_ERROR, "\nCommand should not consist of any args\n");
+		return;
+	}
+
```
Comment (agkaminski) [inline cmd.c:883]: "Command foobar does not take any arguments" perhaps?

We might ignore any additional arguments, shouldn't do us any harm.

File: cmd.c [hunk 4]
```diff
@@ -843,6 +843,75 @@ void cmd_syspage(char *s)
...
+	char args[3][LINESZ + 1];
+
+	for (i = 0; argsc < 3; ++i) {
+		if (cmd_getnext(s, &pos, ". \t", NULL, args[i], sizeof(args[i])) == NULL || *args[i] == 0)
+			break;
+		argsc++;
+	}
+
+	if (argsc != 2) {
+		plostd_printf(ATTR_ERROR, "\nWrong arguments!!\n");
```
Comment (agkaminski) [inline cmd.c:901]: "Wrong number of arguments" perhaps?

File: hal/armv7a/zynq7000/cmd.c [hunk 5]
```diff
@@ -70,5 +64,4 @@ void cmd_bitstream(char *s)
 	}
 
 	plostd_printf(ATTR_LOADER, "PL was successfully initialized\n");
-#endif
-}
+}
```
Comment (agkaminski) [inline hal/armv7a/zynq7000/cmd.c:67]: No new line at the end of the file

**Review by agkaminski — CHANGES_REQUESTED** [review]

File: cmd.c (hunk 2 above)
Comment (HBuczynski) [reply cmd.c:861]: As you can see in `for` loop there is incrementation of `argsc`. We don't  know how many arguments is in the command so firstly you have to parse them and after that check the count.

File: cmd.c (hunk 3 above)
Comment (HBuczynski) [reply cmd.c:883]: I assume that passing wrong arguments is an incorrect usage of command.

File: cmd.c (hunk 2 above)
Comment (agkaminski) [reply cmd.c:861]: Yeah, you're right, I missed that. There is no real need for `i` in this loop. How about:

```
for (argsc  = 0; argsc < 5; ++argsc ) {
	if (cmd_getnext(s, &pos, ". \t", NULL, args[argsc], sizeof(args[argsc])) == NULL || *args[argsc] == '\0')
		break;
}
```

File: cmd.c (hunk 3 above)
Comment (agkaminski) [reply cmd.c:883]: Ok, I have no problem with that ;)

File: cmd.c [hunk 6]
```diff
@@ -724,86 +703,77 @@ void cmd_app(char *s)
...
-			phfs.dataOffs = plostd_ahtoi(appData[i]);
+			default:
+				break;
 		}
-		else if (i == 2) {
-			if (plostd_ishex(appData[i]) < 0) {
-				plostd_printf(ATTR_ERROR, "\nSize is not a hex value !!\n");
-				return;
-			}
+	}
```
[the remark this answers is not in the document]
Comment (HBuczynski) [reply cmd.c:720]: App command can look as follow:
1. `app usb0 psh`
2. `app flash0 @psh`
3. `app flash0 psh(100:100)`

This loop is needed. In the first and second case presented above, only `cmdline` is passed in command, after that the `for` loop will be broken.
In the third case, all parameters will be parsed.

The `switch` statement can be optimized, I will push changes.

File: cmd.c [hunk 7]
```diff
@@ -384,89 +375,81 @@ void cmd_write(char *s)
 }
 
 
-/* Function parse device name or parameters. There is distinction whether it is storage or external dev.
- * For external dev, name shall be provided. For storage device offset and size shall be provided. */
-#if 0
-static int cmd_parseDev(phfs_conf_t *dev, const char (*args)[LINESZ + 1], u16 *argsID, u16 cmdArgsc)
+static int cmd_parseDev(handler_t *h, addr_t *offs, size_t *sz, const char (*args)[LINESZ + 1], u16 *argsID, u16 argsc)
```
[the remark this answers is not in the document]
Comment (HBuczynski) [reply cmd.c:380]: The matrix has to be passed in this way as an argument to the function.

File: cmd.c [hunk 8]
```diff
@@ -384,89 +375,81 @@ void cmd_write(char *s)
...
 {
-	if (plostd_ishex(args[++(*argsID)]) < 0) {
-		plostd_printf(ATTR_LOADER, "\nOpening device %s ..\n", cmd_common.devs[dev->dn].name);
-		dev->handle = phfs_open(dev->dn, args[(*argsID)], 0);
+	const char *alias = args[(*argsID)++];
+
+	if ((*argsID) >= argsc) {
```
[the remark this answers is not in the document]
Comment (HBuczynski) [reply cmd.c]: Done

File: cmd.c (hunk 1 above)
Comment (HBuczynski) [reply cmd.c]: Yes, you have right. I have added information about failure.

File: cmd.c (hunk 2 above)
Comment (HBuczynski) [reply cmd.c]: Done

File: cmd.c (hunk 4 above)
Comment (HBuczynski) [reply cmd.c]: Done

File: hal/armv7a/zynq7000/cmd.c (hunk 5 above)
Comment (HBuczynski) [reply hal/armv7a/zynq7000/cmd.c]: Done

**Review by agkaminski — APPROVED** [review]

**Pull request merged**